crates/node/core/src/node_config.rs

```diff
@@ -138,6 +138,30 @@ impl NodeConfig<ChainSpec> {
     }
 }
 
+// Note: [`From`] trait is better, but it will be conflicting implementation (`impl<T> From<T>`)
+impl<ChainSpec: EthChainSpec + 'static> NodeConfig<ChainSpec> {
```

Comment:
let's not pollute the `reth-node-core` crate with this ExEx-specific code, and instead have this conversion in `crates/exex/exex/src/dyn_context.rs`